R/test.data.table.R

```diff
@@ -60,6 +60,13 @@ test.data.table <- function(verbose=FALSE, pkg="pkg", silent=FALSE, with.other.p
   cat("Running", fn, "\n")
   env = new.env(parent=.GlobalEnv)
   assign("testDir", function(x) file.path(fulldir, x), envir=env)
+
+  # are R's messages being translated to a foreign language? #3039, #630
+  txt = eval(parse(text="tryCatch(mean(not__exist__), error = function(e) e$message)"), envir=.GlobalEnv)
+  foreign = txt != "object 'not__exist__' not found"
+  if (foreign) cat("\n**** This R session's language is not English. Each test will still check that the correct number of errors and/or\n",
```

Comment:
nice idea. excellent.

Reply:
Great. I did try (again) to change the language to English from within test.data.table() but it quickly got complicated and OS-specific. Plus with the link you found about caching the translations. Yep so I figured it would be cleaner to inform user how to change their language and rerun if they want the full-strength test, otherwise just checking the number of errors/warnings is still pretty fine.